src/platform/platform_worker.c

```diff
@@ -300,10 +308,20 @@ CXPLAT_THREAD_CALLBACK(CxPlatWorkerThread, Context)
 #endif
 
         if (Worker->DatapathEC) {
-            CxPlatDataPathRunEC(&Worker->DatapathEC, Worker->ThreadId, WaitTime);
+            if (CxPlatDataPathRunEC(&Worker->DatapathEC, Worker->ThreadId, WaitTime)) {
+                NoWorkCount = 0;
+            }
         } else if (WaitTime != 0) {
             CxPlatEventWaitWithTimeout(Worker->WakeEvent, WaitTime);
+            NoWorkCount = 0;
+        }
+
+        #ifdef _WIN32
+        if (NoWorkCount > 10) {
```

Comment:
Why don't we just yield once when the current loop iteration did not do work?

Reply:
The thought was that doing this too often would hurt perf too much.